src/Symfony/Component/Emoji/Resources/bin/build.php

```diff
@@ -145,16 +142,15 @@ public static function buildGitHubRules(array $emojisCodePoints): iterable
                 continue;
             }
             $emoji = $emojisCodePoints[$emojiCodePoints];
-            if (!self::testEmoji($emoji, 'github', $emojiCodePoints)) {
-                continue;
-            }
             $codePointsCount = mb_strlen($emoji);
             $maps[$codePointsCount][$emoji] = ":$shortCode:";
         }
 
-        $maps = self::createRules($maps);
+        $maps = self::createRules($maps, true);
+        $reverse = array_flip($maps);
+        uksort($reverse, 'strnatcasecmp');
```

Comment:
Is this for strtr or something else ?

If "only" for strtr,  the order is done by the function (cf https://www.php.net/manual/fr/function.strtr.php)

Reply:
Yep it is, at least it was needed when we used the transliterator to do the mapping. Now that we use strtr, we could remove this complexity, but that'd mean recomputing all the json. Better not to keep the git history "light".